crates/gwyneth/src/engine_api.rs

```diff
@@ -119,12 +119,16 @@ pub trait RpcServerArgsExEx {
 impl RpcServerArgsExEx for RpcServerArgs {
     fn with_static_l2_rpc_ip_and_port(mut self, chain_id: u64) -> Self {
         self.http = true;
-        // On the instance the program is running, we wanna have 10111 exposed as the (exex) L2's
-        // RPC port.
         self.http_addr = Ipv4Addr::new(0, 0, 0, 0).into();
-        self.http_port = 10110u16;
-        self.ws_port = 10111u16;
+
+        // Calculate HTTP and WS ports based on chain_id
+        let port_offset = ((chain_id - 167010) / 100000) as u16;
+        self.http_port = 10110 + (port_offset * 10000);
+        self.ws_port = 10111 + (port_offset * 10000);
```

Comment:
these ports are u16, so max value only roughly 65k, so this will fail after 5ish L2s

Reply:
Ok, now there is no chain id offset (167010, 167011, etc..) and ports are only offseted by 100. So that 550ish can fit in.